### PR TITLE
feat use zig cc to target glibc 2.17

### PR DESCRIPTION
**Pull request by lewis6991**
- Does not work with `Debug` or `RelWithDebInfo`

- [x] Test it actually works
   - https://github.com/neovim/neovim-releases/actions/runs/9820554858/job/27115268509

### CODE REVIEW

**Comment by clason** [comment]
If we're building on the host, we can probably avoid the cmake hackery and just use the version ubuntu-latest comes with. @dundargoc 

(The "prepare container" step can also be removed.)

**Comment by lewis6991** [comment]
> (The "prepare container" step can also be removed.)

Completely? Don't we still need to install those packages even on host?

**Comment by clason** [comment]
You're already installing most of them later; just add the missing one (libfuse) there?

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -13,23 +13,21 @@ on:
...
+  ZIG_VERSION: 0.13.0
   BIN_DIR: ${{ github.workspace }}/bin
-  MINIMUM_CMAKE_VERSION: '3.28'
 
 # Build on the oldest supported images, so we have broader compatibility
 # Build with gcc-10 to prevent triggering #14150 (default is still gcc-9 on 20.04)
 jobs:
   linux:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-24.04
```
Comment (clason) [inline .github/workflows/release.yml]:
```suggestion
    runs-on: ubuntu-latest
```
(24.04 is still in beta)

Comment (lewis6991) [reply .github/workflows/release.yml]: 24.04 comes with cmake 3.28 (our min version). 23.10 comes with cmake 3.27

So the choice is use a beta runner, or cmake install hackery.

Comment (clason) [reply .github/workflows/release.yml]: Minimum version is 3.13 tho. (Not sure why this workflow specified 3.28; I assume it was just "if we have to install manually, might at least use the latest".)

Comment (lewis6991) [reply .github/workflows/release.yml]: ah ok. The MINIMUM_CMAKE_VERSION var lead me to believe it was 3.28

**Comment by lewis6991** [comment]
Release is smaller now https://github.com/neovim/neovim-releases/releases/tag/nightly-test

EDIT: forget to restore RelWithDebInfo. Now it's the same.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -13,33 +13,29 @@ on:
...
         run: |
-          apt-get update
-          apt-get install -y software-properties-common
-          add-apt-repository -y ppa:ubuntu-toolchain-r/test # For gcc-10.
-          add-apt-repository -y ppa:git-core/ppa # For git>=2.18.
-          apt-get update
-          apt-get install -y git gcc-10
-          apt-get install -y fuse libfuse2 # For linuxdeploy.
+          sudo apt-get update
+          sudo apt-get install -y git
```
Comment (dundargoc) [inline .github/workflows/release.yml]: Probably not needed.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -51,29 +47,28 @@ jobs:
         with:
           repository: 'neovim/neovim'
           fetch-depth: 0
-      - run: |
-         apt-get update
-         apt-get install -y build-essential curl gettext ninja-build unzip
+
+      # zig cc with -02 implicitly adds -DNDEBUG so remove this from the generator flags
+      - name: Patch Neovim
+        run: sed -i '/APPEND gen_cflags -O2/d' src/nvim/CMakeLists.txt
```
Comment (dundargoc) [inline .github/workflows/release.yml]: You can remove this after merging the PR on neovim repo

**Review by dundargoc — APPROVED** [review]
Looks good, nice work.

Don't forget to update the README to reflect this

File: .github/workflows/release.yml (hunk 3 above)
Comment (lewis6991) [reply .github/workflows/release.yml]: We'll need it to respin 0.10

Comment (dundargoc) [reply .github/workflows/release.yml]: Huh, fair. Backport the neovim pr?

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -13,93 +13,94 @@ on:
...
         uses: actions/checkout@v3
         with:
           repository: 'neovim/neovim'
           fetch-depth: 0
-      - run: |
-         apt-get update
-         apt-get install -y build-essential curl gettext ninja-build unzip
+
+      # zig cc with -02 implicitly adds -DNDEBUG so remove this from the generator flags
+      # Note needed for 0.10.1+ : https://github.com/neovim/neovim/pull/29599
```
Comment (clason) [inline .github/workflows/release.yml]:
```suggestion
      # Not needed for 0.10.1+ : https://github.com/neovim/neovim/pull/29599
```
Or `TODO(lewis): remove after 0.10.1 release`

**Comment by lewis6991** [comment]
All releases updated successfully. Merging.

**Comment by dundargoc** [comment]
Nice work! This will unblock a few other improvements as well.

**Pull request merged**